.github/workflows/cicd.yml

```diff
@@ -0,0 +1,80 @@
+﻿name: 'Continuous Integration'
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+env:
+    DOTNET_CLI_TELEMETRY_OPTOUT: 1
+    DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+    DOTNET_NOLOGO: 1
+  
+jobs:
+  commitlint:
+    name: 📝 Commitlint
+    uses: ./.github/workflows/steps.commitlint.yml
+    secrets: inherit
+      
+  version:
+    name: Define Version
+    uses: ./.github/workflows/steps.dotnet-version.yml
+    with:
+      runs-on: ubuntu-latest
+    secrets: inherit
+
+  build_test:
+    name: 🔨 Build and test
+    needs:
+      - commitlint
+      - version
+    uses: ./.github/workflows/steps.dotnet-build-test.yml
+    with:
+      runs-on: ubuntu-latest
+      version: ${{ needs.version.outputs.version }}
+      publish-package: true
+      use-sonarcloud: false
```

Comment:
We actually have a Sonar Cloud subscription as a sponsored project. Would be interested to see how to use Sonar on this later on.